spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'json'
+
+RSpec.describe Api::V1::UsersController, type: :controller do
+  fixtures :users
+  let(:user) { users(:one) }
+
+  describe 'GET #show' do
+    context 'when is succesful' do
+      before { get :show, params: { id: user.id } }
+
+      it 'return succesful' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns the user' do
+        # Test to ensure response contains the correct email
+        expect(JSON.parse(response.body)).to include('email' => user.email)
+      end
+    end
+
+    context 'when is not succesful' do
+      before { get :show, params: { id: 0 } }
+
+      it 'returns not found' do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'returns the error message' do
+        expect(response.body).to include('User not found')
```

Comment:
I prefer here to have the exactly field with error instead of use include. I mean something like
```
expect(response.body.error).to eql('User not found')
```